streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -93,16 +96,6 @@ public RecordCollectorImpl(final TaskId taskId,
 
         producer = producerSupplier.get(taskId);
 
-        // TODO K9113: this should be moved to task when it transits to running from created / restoring
-        // then even if there's a long time between the initialization and the first txn that is also fine.
-        initialize();
```

Comment:
After further reading on the source code, the original concern of KAFKA-6639 is only for beginTxn, not initTxn, so always calling it right after producer construction is fine (txn.id expiration is 7 days by default). cc @abbccdda, @mjsax.